### PR TITLE
Integrates experiment and evaluation table for the phasar globals analysis evalution

### PR DESCRIPTION
**Pull request by vulder**
Integrates experiment and evaluation table for the globals paper evalution. The experiment evaluates the state with and without the new phasar globals support. It measures the execution time impact of the global analysis support as well as the impact on the generated facts for a linear constant analysis.

### CODE REVIEW

File: varats/varats/data/reports/globals_report.py [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+                if line.startswith("----"):
+                    run_blobs.append(tmp_blob)
+                    tmp_blob = ""
+                else:
+                    tmp_blob += line
+
+        self.__data_from_first = json.loads(run_blobs[0])
+        del self.__data_from_first["runtime-in-seconds"]
+
+        # Calculate timeings
```
Comment (boehmseb) [inline varats/varats/data/reports/globals_report.py]:
```suggestion
        # Calculate timings
```
;)

File: varats/varats/data/reports/globals_report.py [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+            name for the report file that can later be uniquly identified
+        """
+        return ReportFilename.get_file_name(
+            GlobalsReportWithout.SHORTHAND, project_name, binary_name,
+            project_version, project_uuid, extension_type,
+            GlobalsReportWithout.FILE_TYPE
+        )
+
+
+class GlobalsReportWith(GlobalsReport, BaseReport):
```
Comment (boehmseb) [inline varats/varats/data/reports/globals_report.py:175]: Same as with "Without"

File: varats/varats/data/reports/globals_report.py [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+        self.__update_run_values()
+
+    def __str__(self) -> str:
+        output_data = dict(self.__data_from_first)
+        output_data["mean time"] = f"{self.runtime_in_secs.mean} " +\
+            f"(+/- {self.runtime_in_secs.stddev:.2f})"
+        return json.dumps(output_data, indent=4)
+
+
+class GlobalsReportWithout(GlobalsReport, BaseReport):
```
Comment (boehmseb) [inline varats/varats/data/reports/globals_report.py:134]: Without what? Kinda confusing class name here.

File: varats/varats/tools/research_tools/phasar.py [hunk 4]
```diff
@@ -230,4 +230,21 @@ def add_container_layers(
         Args:
             image_context: the base image creation context
         """
-        raise NotImplementedError("Implement phasar container install.")
+        if not self.verify_install(self.install_location()):
+            raise AssertionError(
+                "Phasar is not correctly installed on your system."
+            )
+
+        container_vara_dir = image_context.varats_root / "tools/phasar"
```
Comment (boehmseb) [inline varats/varats/tools/research_tools/phasar.py]: Rename to `container_phasar_dir`

**Review by boehmseb — CHANGES_REQUESTED** [review]

File: varats/varats/data/reports/globals_report.py (hunk 2 above)
Comment (vulder) [reply varats/varats/data/reports/globals_report.py:175]: Well, the experiment specifically compares our analysis with/without globals support. In that context it's quite clear what they refer to.

Comment (boehmseb) [reply varats/varats/data/reports/globals_report.py:175]: Sure, but Reports are used in other places than the experiment (e.g., Plots). And then, it's not immediately obvious.

**Review by ghost — APPROVED** [review]

**Review by boehmseb — APPROVED** [review]

**Pull request merged**